docs: replaced github links with the pkgdown links (#517)
insightsengineering/teal.modules.clinical: README.md

## Acknowledgment

> This package is a result of a joint efforts by many developers and stakeholders. We would like to thank everyone who contributed so far!

Reviewer: ```suggestion
This package is the result of the joint efforts of many developers and stakeholders. We would like to thank everyone who has contributed so far!
```
